src/main/java/org/opensrp/web/rest/UploadController.java

```diff
@@ -0,0 +1,378 @@
+
+	private MultimediaService multimediaService;
+
+	private UploadService uploadService;
+
+	private MultimediaRepository multimediaRepository;
+
+	private OpenmrsIDService openmrsIDService;
+
+	private ClientService clientService;
+
+	private EventService eventService;
+
+	@Autowired
+	public void setMultimediaService(MultimediaService multimediaService) {
+		this.multimediaService = multimediaService;
+	}
+
+	@Autowired
+	public void setUploadService(UploadService uploadService) {
+		this.uploadService = uploadService;
+	}
+
+	@Autowired
+	public void setMultimediaRepository(MultimediaRepository multimediaRepository) {
+		this.multimediaRepository = multimediaRepository;
+	}
+
+	@Autowired
+	public void setOpenmrsIDService(OpenmrsIDService openmrsIDService) {
+		this.openmrsIDService = openmrsIDService;
+	}
+
+	@Autowired
+	public void setClientService(ClientService clientService) {
+		this.clientService = clientService;
+	}
+
+	@Autowired
+	public void setEventService(EventService eventService) {
+		this.eventService = eventService;
+	}
+
+	private List<Map<String, String>> readCSVFile(MultipartFile file) throws IOException {
+		List<Map<String, String>> csv_clients = new ArrayList<>();
+		CSVParser parser;
+		try (Reader reader = new InputStreamReader(file.getInputStream())) {
+			parser = new CSVParser(reader, CSVFormat.DEFAULT.withHeader());
+			List<CSVRecord> records = parser.getRecords();
+			for (CSVRecord record : records) {
+				csv_clients.add(record.toMap());
+			}
+			parser.close();
+		}
+		return csv_clients;
+	}
+
+	@RequestMapping(headers = { "Accept=multipart/form-data" }, method = POST, produces = {
+			MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<String> uploadCSV(@RequestParam("event_name") String eventName,
+			@RequestParam("file") MultipartFile file, HttpServletRequest request) {
+		String teamID = getStringFilter(TEAM_ID, request);
+		String teamName = getStringFilter(TEAM_NAME, request);
+		String locationID = getStringFilter(LOCATION_ID, request);
+
+		String entityId = UUID.randomUUID().toString();
+		String providerId = getCurrentUser();
+		MultimediaDTO multimediaDTO = new MultimediaDTO(entityId.trim(), providerId, file.getContentType().trim(), null,
+				FILE_CATEGORY);
+
+		String status = null;
+		try {
+			List<Map<String, String>> csv_clients = readCSVFile(file);
+
+			UploadValidationBean validationBean = uploadService.validateFieldValues(csv_clients, eventName, globalID);
+			if (validationBean.getErrors() != null && validationBean.getErrors().size() > 0) {
+				validationBean.setAnalyzedData(null);
+				Map<String, Object> map = new HashMap<>();
+				map.put("status", "A number of errors were found during validation");
+				map.put("summary", validationBean);
+				return new ResponseEntity<>(gson.toJson(map), HttpStatus.BAD_REQUEST);
+			}
+
+			UniqueIDProvider uniqueIDProvider = new OpenMRSUniqueIDProvider(openmrsIDService,
```

Comment:
This should be autowired to avoid tight coupling and also to avoid creating every time